tests/testthat/test_tissue_specificity_score.R

```diff
@@ -0,0 +1,85 @@
+context("tissue specificity score")
+
+## create data
+## ... synthetic matrix
+set.seed(42)
+x <- rbind(matrix(runif(n = 90, min = 0, max = 100), ncol = 5),
+           rep(c(0,100), c(4, 1)),
+           rep(c(100,0), c(1, 4)))
+x.NA <- x; x.NA[sample(length(x), size = 5)] <- NA
+x2 <- x[, rep(seq.int(ncol(x)), each = 2)]
+se <- SummarizedExperiment::SummarizedExperiment(list(e = x))
+sce <- as(se, "SingleCellExperiment")
+
+
+test_that("specificityScore() works properly", {
+    ## these should fail
+    expect_error(specificityScore("not_a_valid_x"))
+    expect_error(specificityScore(x, method = "not_a_valid_method"))
+    expect_error(specificityScore(x, "counts", group = "error"))
+    expect_error(specificityScore(x, "counts", thresh = "error"))
+    expect_error(specificityScore(x, "counts", thresh = 1:2))
+    expect_error(specificityScore(x, "counts", na.rm = c(TRUE,FALSE)))
+    expect_error(specificityScore(x, "counts", na.rm = "error"))
+    expect_error(specificityScore(se, expr_values = 3))
+    expect_error(specificityScore(se, expr_values = "non_existing"))
+    expect_error(specificityScore(se, expr_values = TRUE))
+    expect_error(specificityScore(sce, expr_values = 3))
+    expect_error(specificityScore(sce, expr_values = "non_existing"))
+    expect_error(specificityScore(sce, expr_values = TRUE))
+    
+    ## run function
+    ## ... tau
+    res.tau1 <- specificityScore(x, method = "tau")
+    res.tau2 <- specificityScore(se, method = "tau", expr_values = 1)
+    res.tau3 <- specificityScore(se, method = "tau", expr_values = "e")
+    res.tau4 <- specificityScore(sce, method = "tau", expr_values = 1)
+    res.tau5 <- specificityScore(sce, method = "tau", expr_values = "e")
+    res.tau6 <- specificityScore(x.NA, method = "tau")
+    res.tau7 <- specificityScore(x.NA, method = "tau", na.rm = TRUE)
+    res.tau8 <- specificityScore(x2, method = "tau",
```

Comment:
Worth adding a test with differently sized groups?

Reply:
Yes, good point, I added such a text (`x3` and `res.tau9`)